src/installer/installer.go

```diff
@@ -172,6 +172,13 @@ func (i *installer) InstallNode() error {
 			return err
 		}
 		i.log.Info("Setting bootstrap node new role to master")
+
+		// In the agent-based installer the assisted-service runs on the bootstrap node, so let's
+		// wait for the workers (if any) to kick-off the joining process before rebooting the
+		// bootstrap node.
+		if err = i.waitForWorkers(ctx); err != nil {
```

Comment:
this is relevant only for ABI no ? There is no point to wait for worker in regular flow

Reply:
found https://github.com/openshift/assisted-installer/pull/910/files#diff-1046cc2d18cf5f82336bbad36a2d28540606e1c6aaa0b5073c545301ef60ffd4R201